tls/s2n_x509_validator.c

```diff
@@ -616,6 +619,14 @@ S2N_RESULT s2n_x509_validator_validate_cert_chain(struct s2n_x509_validator *val
         RESULT_GUARD_POSIX(s2n_extension_list_process(S2N_EXTENSION_LIST_CERTIFICATE, conn, &first_certificate_extensions));
     }
 
+    if (conn->config->cert_validation_cb) {
+        struct s2n_cert_validation_info info = { 0 };
+        int ret = conn->config->cert_validation_cb(conn, &info, conn->config->cert_validation_ctx);
+        RESULT_ENSURE(ret == S2N_SUCCESS, S2N_ERR_CANCELLED);
+        RESULT_ENSURE(info.finished, S2N_ERR_INVALID_STATE);
```

Comment:
Returning from the callback without calling accept or reject is only an implicit reject because the callback isn't asynchronous. If we later decide to make this function asynchronous, we would return blocked instead of failing.

Reply:
Okay I think I get it. The people who rely on the implicit reject mechanism will be blocked indefinitely if/when we make this async in the future.

Reply:
Right exactly, though I'm not sure how much of a difference changing the error will make to prevent this, so maybe it's not worth checking.

Reply:
Eh, you're right, I don't know if an invalid state error will prevent someone from just not calling either accept or reject when they want to reject. It's worth a shot though.